packages/bsky/tests/labeler/labeler.test.ts

```diff
@@ -0,0 +1,205 @@
+        constraints: {},
+      }
+      await repoSvc.blobs.verifyBlobAndMakePermanent(alice, preparedBlobRef)
+      await repoSvc.blobs.associateBlob(
+        preparedBlobRef,
+        postUri(),
+        await cidForCbor(1),
+        alice,
+      )
+      return blobRef
+    }
+    const bytes1 = new Uint8Array([1, 2, 3, 4])
+    const bytes2 = new Uint8Array([5, 6, 7, 8])
+    const bytes3 = new Uint8Array([4, 3, 2, 1])
+    badBlob1 = await storeBlob(bytes1)
+    badBlob2 = await storeBlob(bytes2)
+    goodBlob = await storeBlob(bytes3)
+  })
+
+  afterAll(async () => {
+    await testEnv.close()
+  })
+
+  it('labels text in posts', async () => {
+    const post = {
+      $type: 'app.bsky.feed.post',
+      text: 'blah blah label_me',
+      createdAt: new Date().toISOString(),
+    }
+    const cid = await cidForRecord(post)
+    const uri = postUri()
+    labeler.processRecord(uri, post)
+    await labeler.processAll()
+    const labels = await labelSrvc.getLabels(uri.toString())
+    expect(labels.length).toBe(1)
+    expect(labels[0]).toMatchObject({
+      src: labelerDid,
+      uri: uri.toString(),
+      cid: cid.toString(),
+      val: 'test-label',
+      neg: false,
+    })
+  })
+
+  it('labels embeds in posts', async () => {
+    const post = {
+      $type: 'app.bsky.feed.post',
+      text: 'blah blah',
+      embed: {
+        $type: 'app.bsky.embed.images',
+        images: [
+          {
+            image: badBlob1,
+            alt: 'img',
+          },
+          {
+            image: badBlob2,
+            alt: 'label_me_2',
+          },
+          {
+            image: goodBlob,
+            alt: 'img',
+          },
+        ],
+      },
+      createdAt: new Date().toISOString(),
+    }
+    const uri = postUri()
+    labeler.processRecord(uri, post)
+    await labeler.processAll()
+    const dbLabels = await labelSrvc.getLabels(uri.toString())
+    const labels = dbLabels.map((row) => row.val).sort()
+    expect(labels).toEqual(
+      ['test-label', 'test-label-2', 'img-label', 'other-img-label'].sort(),
+    )
+  })
+
+  it('labels text & imgs in profiles', async () => {
+    const profile = {
+      $type: 'app.bsky.actor.profile',
+      displayName: 'label_me',
+      description: 'label_me_2',
+      avatar: badBlob1,
+      banner: badBlob2,
+      createdAt: new Date().toISOString(),
+    }
+    const uri = profileUri()
+    labeler.processRecord(uri, profile)
+    await labeler.processAll()
+    const dbLabels = await labelSrvc.getLabels(uri.toString())
+    const labels = dbLabels.map((row) => row.val).sort()
+    expect(labels).toEqual(
+      ['test-label', 'test-label-2', 'img-label', 'other-img-label'].sort(),
+    )
+  })
+
+  it('retrieves both profile & repo labels on profile views', async () => {
+    await ctx.db.db
+      .insertInto('label')
+      .values({
+        src: labelerDid,
+        uri: alice,
+        cid: '',
+        val: 'repo-label',
+        neg: 0,
```

Comment:
another place that should be a bool

Reply:
Thanks for surfacing!  Now that it's in the db model typescript's got my back too 😎 So these issues should be pretty well taken care of.